README.md

```diff
@@ -33,52 +33,92 @@ in the x-common repository.  This describes how all the language tracks are put
+directory, you'll see there are three files there; *example.go*, *hello_test.go*,
+and *hello_world.go*.
+
+```sh
+~/exercism/go/hello-world
+$ tree
+.
+├── example.go
+├── hello_test.go
+└── hello_world.go
+
+0 directories, 3 files
+```
+
+### Example solutions
+
+*example.go* is a reference solution.  It is a valid solution that [Travis](https://travis-ci.org/exercism/xgo),
+the CI (continuous integration) service, can run tests against.  Solvers generally
+will not see it though.  Files with *"example"* in the file name are skipped by
+the `exercism fetch` command.  Because of this, there is less need for this code
+to be a model of style, expression and readability, or to use the best algorithm.
+Examples can be plain, simple, concise, even naïve, as long as they are correct.
+The test file though, is fetched for the solver and deserves attention for consistency
+and appearance.
+
+### Tests
+
+*hello_test.go* uses a data-driven test. Test cases are defined as data, then a
+test function iterates over the data. Identifiers within the method appear in
+actual-expected order as described at [Useful Test Failures](https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures).
+Here the identifier *"observed"* is used instead of actual. That's fine. More
+common are words *"got"* and *"want"*. They are clear and short. Note
+[Useful Test Failures](https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures)
+is part of [Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments).
+Really we like most of the advice on that page.
+
+In Go we generally have all tests enabled and do not ask the solver to edit the
+test program, to enable progressive tests for example. `t.Fatalf`, as seen
+in [*hello_test.go*](https://github.com/exercism/xgo/blob/master/exercises/hello-world/hello_test.go#L26),
+will stop tests at the first failure encountered, so the solver is not faced with
+too many failures all at once.
+
+We like errors in Go. It's not idiomatic Go to ignore invalid data or have undefined
+behavior. Sometimes our Go tests require an error return where other language
+tracks don't.
+
+### Benchmarks
+
+In most test files there will also be benchmark tests, for example in the
+[clock exercise](https://github.com/exercism/xgo/blob/master/exercises/clock/clock_test.go#L75-L91).
+In Go, benchmarking is a first-class citizen of the testing package. We throw in
+benchmarks because they're interesting, and because it is idiomatic in Go to think
+about performance. There is no critical use for these though. Usually they will
+just bench the combined time to run over all the test data rather than attempt
+precise timings on single function calls. They are useful if they let the solver
+try a change and see a performance effect. It is important, though, to keep in
+mind that benchmark tests are reflective of the machine they are run on. Benchmark
```

Comment:
No, you're right. I wanted to include something somewhere, and started in the README, but then realised it would be beneficial in the track_hints file, and just left it here. The information is more for users than someone adding an exercise/test.